### DISCUSSION TITLE
Meta: Improve Cross-Browser Compatibility

### DISCUSSION THREAD

**Discussion by amacneil** [foxglove/community#174] [Q&A]
This is a meta-issue tracking work happening to improve cross-browser compatibility. Currently, we only support Chrome (v76+?).

- [x] Edge: I don't believe anyone has tested, but since it uses Blink it might work?
- [ ] Safari, and all browsers on iOS/iPadOS (which are forced to use Webkit) (https://github.com/foxglove/community/issues/149)
- [ ] Firefox
- [ ] https://github.com/foxglove/studio/issues/2505

**Comment by ic** [comment]
I would like to just share the latest container to this day does not work at all on Firefox. It has become such a minor browser that it may not be relevant, but main one here.

The error trace is little interesting, blocked at initialisation.

<img width="1086" alt="image" src="https://user-images.githubusercontent.com/64160/149708036-d76064e0-d469-4caf-8f5a-b93c30fba40f.png">

Also, I am on an beta version of the Firefox (97.0b2 (64-bit)), not always stable, yet everything else seems to work as expected.

Good luck with all the tasks ahead (beyond browser support!)

**Comment by amacneil** [comment]
We don't support Firefox yet, since it's lacking a few features and we have not yet built workarounds (also, we have not yet set up cross-browser testing which is a pre-requisite to proper support).

However, the error you're seeing is weird. It looks like maybe related to an extension you have installed? I can visit [studio.foxglove.dev](https://studio.foxglove.dev) in FF (Ubuntu 20.04 ARM) just fine. Does that only happen in the container for you or does it also happen on [studio.foxglove.dev](https://studio.foxglove.dev)?

**Comment by jtbandes** [comment]
Per instructions in the readme, I just tried running `docker run -it --rm -p "8080:8080" ghcr.io/foxglove/studio:latest` and visiting localhost:8080 in Firefox 96.0.1, and it works for me. What data source are you loading into Studio?

**Comment by ic** [comment]
> We don't support Firefox yet, since it's lacking a few features and we have not yet built workarounds (also, we have not yet set up cross-browser testing which is a pre-requisite to proper support).
> 
> However, the error you're seeing is weird. It looks like maybe related to an extension you have installed? I can visit [studio.foxglove.dev](https://studio.foxglove.dev) in FF (Ubuntu 20.04 ARM) just fine. Does that only happen in the container for you or does it also happen on [studio.foxglove.dev](https://studio.foxglove.dev)?

Thank you @amacneil for your prompt feedback. And sorry for the late reply.

As for today, the online version at studio.foxglove.dev does work fine, indeed! It does point out that the only supported browser is Chrome, but seems functional (not used much at this point).

**Comment by ic** [comment]
> Per instructions in the readme, I just tried running `docker run -it --rm -p "8080:8080" ghcr.io/foxglove/studio:latest` and visiting localhost:8080 in Firefox 96.0.1, and it works for me. What data source are you loading into Studio?

I could not reach the data loading step :-D

If working in 96.x, then a problem with the version I am using (beta 97).

Given your feedback, please discard my comment, as it should be transient issue.

We are a handful on Firefox interested in Studio, and will watch any call for testers.

**Comment by amacneil** [comment]
I think those errors are either happening because you're using a beta version of Firefox, or because of some extension you have installed. I would be curious to hear if it works in the latest stable version of Firefox with no extensions loaded?

**Comment by TBK** [comment]
I can confirm that it works perfectly fine with Microsoft Edge (100.0.1185.44) on Windows 11.

Newer Edge versions are based on Chromium.

**Comment by amacneil** [comment]
Yeah we should probably list Edge as officially supported and remove the warnings, since it's all Chromium under the hood.

**Comment by joewong00** [comment]
Hi, is there any updates on this? Foxglove websocket still cannot be opened on chrome/safari/edge in tablets or phones despite being the latest version. Thanks.

**Reply by jtbandes** [reply]
Hi,
There’s still no official update to share on this. However the issue you’re describing sounds like you are encountering a security error rather than a cross-browser compatibility issue. If you can enable SSL on your websocket server (`wss://`) then it should load in any browser on any device.

**Reply by joewong00** [reply]
Hi, I already tried using SSL on websocket server. Unfortunately I am still unable to connect to the ros master using latest chrome on IOS. Is it really already fully supported on any browser and any device?

**Reply by Jooolee** [reply]
I have same scene and something wrong.

**Comment by Jooolee** [comment]
when resolve this problem? Our team need play bag in Phone..
![image](https://github.com/foxglove/community/assets/37429047/5d5bf83d-0157-4b45-be83-823787ed7188)

in Iphone, webpack-dev-serve  disconnected!  and Stop to 'Setting render frame to world'
![image](https://github.com/foxglove/community/assets/37429047/f65f136f-5657-42f2-bf90-9b20b1c8852f)

In Chrome Web. it is ok.  It can show log 'Setting render frame to World'
![image](https://github.com/foxglove/community/assets/37429047/6d0982a1-4410-41b2-ab8b-dc5813f7a957)

**Reply by Jooolee** [reply]
In Phone safari（webkit），It look like waiting load bag.
